python/cudf/cudf/core/index.py

```diff
@@ -58,11 +58,37 @@ def _to_frame(this_index, index=True, name=None):
 
 
 class Index(Frame, Serializable):
-    def __new__(cls, data=None, dtype=None, name=None, **kwargs):
+    def __new__(
+        cls,
+        data=None,
+        dtype=None,
+        copy=False,
+        name=None,
+        tupleize_cols=True,
+        **kwargs,
+    ):
+        if tupleize_cols is not True:
+            raise NotImplementedError(
+                "tupleize_cols != True is not yet supported"
+            )
 
+        if (
+            copy is True
+            and data is not None
+            and hasattr(data, "cuda_array_interface")
+        ):
+            data = data.copy()
         return as_index(data, dtype=dtype, name=name, **kwargs)
 
-    def __init__(self, data=None, dtype=None, name=None, **kwargs):
+    def __init__(
+        self,
+        data=None,
+        dtype=None,
+        copy=False,
+        name=None,
+        tupleize_cols=True,
+        **kwargs,
+    ):
         """Immutable ndarray implementing an ordered, sliceable set.
         The basic object storing row labels for all cuDF objects.
```

Comment:
I don't think Indexes are really an `ndarray` or a `set`, can we update this?

Reply:
This has been already taken care